crates/llm-ls/src/main.rs

```diff
@@ -613,7 +614,8 @@ impl Backend {
                 "received completion request for {}",
                 params.text_document_position.text_document.uri
             );
-            if params.api_token.is_none() {
+            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(&adaptors::DEFAULT_ADAPTOR.to_string()).as_str() == adaptors::HUGGING_FACE;
+            if params.api_token.is_none() && is_using_hf_inference {
```

Comment:
```suggestion
            if params.api_token.is_none() && is_using_inference_api {
```
Should've added all suggestions sorry!